app/src/gecko/java/org/mozilla/focus/web/WebViewProvider.java

```diff
@@ -96,34 +141,106 @@ public void cleanup() {
+                    contentTrackersBlocked = sharedPreferences.getBoolean(prefName, false);
+                }
+            }
+            updateBlocking();
+        }
+
+        private void updateBlocking() {
+            int categories = 0;
+            if (socialTrackersBlocked) {
+                categories += GeckoSession.TrackingProtectionDelegate.CATEGORY_SOCIAL;
+            }
+            if (adTrackersBlocked) {
+                categories += GeckoSession.TrackingProtectionDelegate.CATEGORY_AD;
+            }
+            if (analyticTrackersBlocked) {
+                categories += GeckoSession.TrackingProtectionDelegate.CATEGORY_ANALYTIC;
+            }
+            if (contentTrackersBlocked) {
+                categories += GeckoSession.TrackingProtectionDelegate.CATEGORY_CONTENT;
+            }
+            if (geckoSession != null) {
+                geckoSession.enableTrackingProtection(categories);
+            }
+        }
+
+        private GeckoSession.ContentListener createContentListener() {
+            return new GeckoSession.ContentListener() {
+                @Override
+                public void onTitleChange(GeckoSession session, String title) {
+                    webViewTitle = title;
+                }
+
                 @Override
-                public void onTitleChange(GeckoView geckoView, String s) {
+                public void onFullScreen(GeckoSession session, boolean fullScreen) {
+                    if (fullScreen) {
+                        callback.onEnterFullScreen(new FullscreenCallback() {
+                            @Override
+                            public void fullScreenExited() {
+                                geckoSession.exitFullScreen();
+                            }
+                        }, null);
+                    } else {
+                        callback.onExitFullScreen();
+                    }
                 }
 
                 @Override
-                public void onFullScreen(GeckoView geckoView, boolean fullScreen) {
+                public void onContextMenu(GeckoSession session, int screenX, int screenY, String uri, String elementSrc) {
+                    if (elementSrc != null && uri != null) {
+                        callback.onLongPress(new HitTarget(true, uri, true, elementSrc));
+                    } else if (elementSrc != null) {
+                        callback.onLongPress(new HitTarget(false, null, true, elementSrc));
+                    } else if (uri != null) {
+                        callback.onLongPress(new HitTarget(true, uri, false, null));
+                    }
+                }
+
+                @Override
+                public void onFocusRequest(GeckoSession geckoSession) {
+
                 }
             };
         }
 
-        private ProgressListener createProgressListener() {
-            return new ProgressListener() {
+        private GeckoSession.ProgressListener createProgressListener() {
```

Comment:
\[PMD\] INFO: The method 'createProgressListener' has a Cyclomatic Complexity of 7\.